main/lsp/request_dispatch.cc

```diff
@@ -65,34 +67,34 @@ LSPResult LSPLoop::processRequestInternal(unique_ptr<core::GlobalState> gs, cons
         if (method == LSPMethod::SorbetWorkspaceEdit) {
             // Note: We increment `lsp.messages.processed` when the original requests were merged into this one.
             auto &editParams = get<unique_ptr<SorbetWorkspaceEditParams>>(params);
-            auto &counts = editParams->counts;
-            prodCategoryCounterAdd("lsp.messages.processed", "textDocument.didChange", counts->textDocumentDidChange);
-            prodCategoryCounterAdd("lsp.messages.processed", "textDocument.didOpen", counts->textDocumentDidOpen);
-            prodCategoryCounterAdd("lsp.messages.processed", "textDocument.didClose", counts->textDocumentDidClose);
-            prodCategoryCounterAdd("lsp.messages.processed", "sorbet/watchmanFileChange",
-                                   counts->sorbetWatchmanFileChange);
-            // Number of messages merged together into a workspace edit.
-            prodCounterAdd("lsp.messages.merged", (counts->textDocumentDidChange + counts->textDocumentDidOpen +
-                                                   counts->textDocumentDidClose + counts->sorbetWatchmanFileChange) -
-                                                      1);
-            return commitTypecheckRun(runTypechecking(move(gs), move(editParams->updates)));
-        }
-        if (method == LSPMethod::Initialized) {
+            const u4 end = editParams->updates.versionEnd;
+            const u4 start = editParams->updates.versionStart;
+            // Versions are sequential and wrap around. Use them to figure out how many edits are contained within this
+            // update.
+            const u4 merged = min(end - start, 0xFFFFFFFF - start + end);
+            auto run = runTypechecking(move(gs), move(editParams->updates));
+            // Only report stats if the edit was committed.
+            if (!run.canceled) {
+                prodCategoryCounterInc("lsp.messages.processed", "sorbet/workspaceEdit");
```

Comment:
I simplified these metrics. Tracking this data at a finer granularity just increases the complexity of the code w/o much benefit.